eventstream_rpc/source/EventStreamClient.cpp

```diff
@@ -487,19 +517,32 @@ namespace Aws
             return onFlushPromise.get_future();
         }
 
-        std::future<RpcError> ClientConnection::Close() noexcept
+        void ClientConnection::Close() noexcept
         {
+            m_stateMutex.lock();
```

Comment:
The pattern here is more easily and safely done via RAII using: https://en.cppreference.com/w/cpp/thread/lock_guard